.github/workflows/docker.yml

```
@@ -0,0 +1,117 @@
+name: Build and Push Dockerimage
+
+on:
+  push:
+    branches: ['main']
+    paths:
+      - '**.c'
+      - '**.h'
+      - Dockerfile
+      - '.github/workflows/docker.yml'
+    tag:
```

actionlint (1.7.4)

unexpected key "tag" for "push" section. expected one of "branches", "branches-ignore", "paths", "paths-ignore", "tags", "tags-ignore", "types", "workflows"

(syntax-check)